src/globus_sdk/services/gcs/data/endpoint.py

```diff
@@ -0,0 +1,146 @@
+        network. Allowed values are:
+
+        * ``normal``: (Default) Uses an average level of concurrency and
+          parallelism. The levels depend on the number of physical servers in the
+          endpoint.
+        * ``minimal``: Uses a minimal level of concurrency and parallelism.
+        * ``aggressive``: Uses a high level of concurrency and parallelism.
+        * ``custom``: Uses custom values of concurrency and parallelism set by the
+          endpoint admin. When setting this level, you must also set the
+          ``max_concurrency``, ``preferred_concurrency``, ``max_parallelism``, and
+          ``preferred_parallelism`` properties.
+    :param organization: Organization that runs the server(s). Searchable. Unicode
+        string, max 1024 characters, no new lines.
+    :param subscription_id: The id of the subscription that is managing this endpoint.
+        This may be the special value DEFAULT when using this as input to PATCH or PUT
+        to use the caller’s default subscription id.
+
+    :param keywords: List of search keywords for the endpoint. Unicode string, max 1024
+        characters total across all strings.
+
+    :param allow_udt: Allow data transfer on this endpoint using the UDT protocol.
+    :param public: Flag indicating whether this endpoint is visible to all other Globus
+        users. If false, only users which have been granted a role on the endpoint or
+        one of its collections, or belong to a domain allowed access to any of its
+        storage gateways may view it.
+
+    :param gridftp_control_channel_port: TCP port for the Globus control channel to
+        listen on. By default, the control channel is passed through 443 with an ALPN
+        header containing the value "ftp".
+    :param max_concurrency: Admin-specified value when the ``network_use`` property’s
+        value is ``"custom"``; otherwise the preset value for the specified
+        ``network_use``.
+    :param max_parallelism: Admin-specified value when the ``network_use`` property’s
+        value is ``"custom"``; otherwise the preset value for the specified
+        ``network_use``.
+    :param preferred_concurrency: Admin-specified value when the ``network_use``
+        property’s value is ``"custom"``; otherwise the preset value for the specified
+        ``network_use``.
+    :param preferred_parallelism: Admin-specified value when the ``network_use``
+        property’s value is ``"custom"``; otherwise the preset value for the specified
+        ``network_use``.
+    """
+
+    DATATYPE_BASE = "endpoint"
+    DATATYPE_VERSION_IMPLICATIONS: dict[str, tuple[int, int, int]] = {
+        "gridftp_control_channel_port": (1, 1, 0),
+    }
+    DATATYPE_VERSION_CALLBACKS: tuple[DatatypeCallback, ...] = ()
+
+    # Note: The fields below represent the set of mutable endpoint fields in
+    #   an Endpoint#1.2.0 document.
+    #   https://docs.globus.org/globus-connect-server/v5.4/api/schemas/Endpoint_1_2_0_schema/
```

Comment:
I'm alert to the fact that I see this schema listed as 1.2.0, but the mapping above only includes 1.1.0.
Is that because 1.2.0 added read-only fields? I want to confirm that this isn't a mistake.

Reply:
Yep; from that doc:
> Version 1.2.0 of the endpoint includes read-only earliest_last_access to put a limit on collections which are missing a last_access value.

There's no mutable fields included in the 1.1.0 -> 1.2.0 version bump.